### PR TITLE
WIP - Consolidate Jobs

### PR DESCRIPTION
**Pull request by sprucely**
This turned out to be bigger than expected, but I don't think I did anything too radical. Probably should make the diff ignore whitespace changes.

I moved the JobRecord stuff out of the AE jobs to be able to treat them like the other jobs when manually launching via the `BackgroundTaskQueue`.

I needed to register `AggieEnterpriseService` and  `AggieEnterpriseOptions` in `Sloth.Web`, so I should probably double-check that those settings are present in Test and Prod.

@jSylvestre Autoformatting knocked out a bunch of those equal-aligned blocks. Let me know if you would like me to fix those.

@cydoval Would you happen to have any stock images for the AE and WebHook jobs? And word-wrap in the cards seems to make alignment a bit wonky...

image

### CODE REVIEW

File: Sloth.Core/Domain/Integration.cs [hunk 1]
```diff
@@ -38,8 +37,7 @@ public class Integration
         [MaxLength(128)]
         public string HoldingAccount { get; set; }
 
-        public IList<CybersourceBankReconcileJobBlob> CybersourceBankReconcileJobBlobs { get; set; }
-
+        public IList<TransactionBlob> TransactionBlobs { get; set; }
```
Comment (jSylvestre) [inline Sloth.Core/Domain/Integration.cs:40]: Not a fan of using the word transaction here. Call it JobBlobs ? JobRecordBlobs?

File: Sloth.Core/Domain/WebHookRequest.cs [hunk 2]
```diff
@@ -30,10 +30,6 @@ public class WebHookRequest
 
         public bool Persist { get; set; }
 
-        public WebHookRequestResendJobRecord WebHookRequestResendJob { get; set; }
```
Comment (jSylvestre) [inline Sloth.Core/Domain/WebHookRequest.cs:33]: Does this need to point to the common one?

File: Sloth.Core/Jobs/CybersourceBankReconcileJob.cs [hunk 3]
```diff
@@ -42,34 +41,36 @@ public CybersourceBankReconcileJob(SlothDbContext context, ICyberSourceBankRecon
                 if (!integrations.Any())
                 {
                     log.Information("Early exit, no active integrations found.");
-                    yield break;
```
Comment (jSylvestre) [inline Sloth.Core/Jobs/CybersourceBankReconcileJob.cs:45]: You don't need this?

File: Sloth.Core/Sloth.Core.csproj [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>net6</TargetFramework>
-    <LangVersion>8.0</LangVersion>
+    <LangVersion>9.0</LangVersion>
```
Comment (jSylvestre) [inline Sloth.Core/Sloth.Core.csproj:5]: latest instead?

File: Sloth.Core/Domain/WebHookRequest.cs (hunk 2 above)
Comment (sprucely) [reply Sloth.Core/Domain/WebHookRequest.cs:33]: Oops. That one needs to be deleted. We're not going to have any foreign keys to the JobRecords table.

File: Sloth.Core/Jobs/CybersourceBankReconcileJob.cs (hunk 3 above)
Comment (sprucely) [reply Sloth.Core/Jobs/CybersourceBankReconcileJob.cs:45]: Not anymore. The `CyberSourceBankReconcileService` now does the work of saving the blob records. So they don't need to be passed up to let the caller save them.

File: Sloth.Core/Domain/Integration.cs (hunk 1 above)
Comment (sprucely) [reply Sloth.Core/Domain/Integration.cs:40]: But it's a new many-to-many between Transactions and Blobs, not JobRecords and Blobs.

**Review by jSylvestre — APPROVED** [review]
Seems ok, didn't run anything.
Might want to get @srkirkland to look at it too.
Then I guess deploy and test.

**Review by srkirkland — APPROVED** [review]

**Review by jSylvestre — APPROVED** [review]

**Pull request merged**